packages/utils/interactive-hooks/src/Checkbox.hooks.ts

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { IKeyboardEvent } from '@office-iss/react-native-win32';
+
+export type onToggleCallback = (value: boolean) => void;
+export type boxSideType = 'start' | 'end' | undefined;
+export type keyUpCallback = () => void;
+export type checkedType = boolean | undefined;
+
+/* Re-usable hook for toggle components.
+ ** This hook configures the checked state, the callback to toggle the component, and configures the boxSide prop/state.
+ ** It handles the controlled/uncontrolled functionality of the toggle component.
+ **
+ ** PROPS:  defaultChecked - Default checked state. Mutually exclusive to ‘checked’. This should come from userProps
+ **         checked - Checked state. Mutually exclusive to 'defaultChecked'. This should come from userProps
+ **         userCallback() - Callback provided by userProps when the checked (toggle) state changes
+ **         boxSide - Allows you to set the toggle to be at the before (start) or after (end) the label. Comes from userProps
+ ** RETURNS:
+ **         onChange() - Callback to toggle the component
+ **         state.isChecked - Whether or not component is currently checked or selected
+ **         state.boxAtEnd - Whether the toggle component is at the end or start
+ */
+export function useAsToggle(defaultChecked: checkedType, checked: checkedType, userCallback?: onToggleCallback, boxSide?: boxSideType) {
+  // NOTE: Using "defaultChecked ?? checked" gives me a parsing error when building
+  const [isChecked, setChecked] = React.useState(defaultChecked != undefined ? defaultChecked : checked);
+
+  const onChange = React.useCallback(() => {
+    userCallback && userCallback(!isChecked);
+    setChecked(!isChecked);
+  }, [isChecked]);
+
+  const boxAtEnd = boxSide == undefined || boxSide == 'start' ? false : true;
+
+  return { onChange, state: { isChecked: checked != undefined ? checked : isChecked, boxAtEnd } };
+}
```

Comment:
From a code reuse perspective, I still think this hook could use some work.

- boxSide doesn't feel like it belongs in the hook
- checkedType makes no sense to me.... it's literally the same as `checked?: boolean`
- I'm not crazy about this return type here... I might make this semantically similar to useState.
  - `return [ isChecked, onChange ]`
  - `const [ checked, toggleChecked ] = useAsToggle`
    - I feel like this makes it a little easier to understand when we assign the toggleChecked fn to the onPress input
    - if you did destructure the array in usePrepareProps as `[ checked,` you'd have to rename the userProps in the destructuring to `userChecked`
- I would prefer to only take one boolean for the checked state I think and let the caller sort out initial value because this hook primarily adds the uncontrolled functionality

Reply:
Well I guess it's doing controlled stuff, too, so it's a bit complicated.  It's kind of an ambiguous user scenario we're trying to design for but it's not hard for me to imagine this getting use elsewhere... It's possible it's best to refactor this when we need it.  I'm curious to hear your thoughts :)

Reply:
I kinda early posted this before I finished typing my thoughts out, but hopefully you get the gist of where my head is at

Reply:
Actually, to make this conversation less abstract and more concrete... look at @PPatBoyd 's Callout PR test page.  It looks  like he's sort of hand rolling some toggle buttons.  What if we wanted to use this hook over there?

Reply:
Thanks for the feedback, here are my thoughts:

- I added boxSide in this hook because I looked into the Toggle switch from Fabric (I'm assuming we'll create this at some point), and they have a prop similar to it that they use for determining if the text should be on the left or right side, so I thought I would add it here. But I do see from a more generalized view that boxSide would not be valid in other scenarios. I'll remove it for now. Thoughts?

- Reverting checkedType to ?: (Still getting used to this :) )

- Since it is doing uncontrolled/controlled stuff, it is a little more complicated. Maybe leave this for now and then we'll come back to it if we find it needs some touching up?

Let me know what you think

Reply:
Yea, to some extent we're trying to design for somewhat abstract scenario (although Pat's test page makes it look a little more concrete), so overengineering this now is a bit of YAGNI violation, you could make the case to do this later.

Controlled/Uncontrolled aside, I do think there's value in aligning with the naming and calling conventions of useState like I talk about in the 3rd bullet.  

I'd be curious to hear what @PPatBoyd  thinks.

It's your call, though, I won't block the PR on this issue

Reply:
I'm going to change the naming to align with useState, I think you're right about that. I'm doing that now and it'll be included in my next PR so I can merge this one now :)